common/requestfilter/sigfilter.go

```diff
@@ -10,80 +10,137 @@ import (
 )
 
 type SigFilter struct {
 	clientSignatureVerificationRequired bool
+	channelID                           string
+	policyManager                       policies.Manager
+	configTxValidator                   configtx.Validator
+	signer                              protoutil.Signer
 }
 
 func NewSigFilter(config FilterConfig) *SigFilter {
-	return &SigFilter{clientSignatureVerificationRequired: config.GetClientSignatureVerificationRequired()}
+	return &SigFilter{
+		clientSignatureVerificationRequired: config.ClientSignatureVerificationRequired(),
+		channelID:                           config.ChannelID(),
+		policyManager:                       config.PolicyManager(),
+		configTxValidator:                   config.ConfigTxValidator(),
+		signer:                              config.Signer(),
+	}
 }
 
 func (sf *SigFilter) Verify(request *comm.Request) error {
-	// extract signedData, while verifying the structure of the request
-	signedData, err := requestToSignedData(request)
+	signedData, isConfig, err := sf.requestToSignedData(request)
 	if err != nil {
 		return fmt.Errorf("failed to convert request to signedData : %s", err)
 	}
-	_ = signedData
 
 	if sf.clientSignatureVerificationRequired {
-		// extract policy and validate the signature - TBD
+		policy, exists := sf.policyManager.GetPolicy(policies.ChannelWriters)
+		if !exists {
+			return fmt.Errorf("no policies in config block")
+		}
+		err = policy.EvaluateSignedData([]*protoutil.SignedData{signedData})
+		if err != nil {
+			return fmt.Errorf("signature did not satisfy policy %s", policies.ChannelWriters)
+		}
+
+		if isConfig {
+			// validate config envelope
+			configEnvelope, err := sf.configTxValidator.ProposeConfigUpdate(&common.Envelope{
+				Payload:   request.Payload,
+				Signature: request.Signature,
+			})
+			if err != nil {
+				return errors.WithMessagef(err, "error applying config update to existing channel '%s'", sf.configTxValidator.ChannelID())
+			}
+
+			config, err := protoutil.CreateSignedEnvelope(common.HeaderType_CONFIG, sf.configTxValidator.ChannelID(), sf.signer, configEnvelope, int32(0), 0)
+			if err != nil {
+				return err
+			}
+
+			envelopeAsSignedData, err := protoutil.EnvelopeAsSignedData(config)
+			if err != nil {
+				return fmt.Errorf("could not convert message to signedData: %s", err)
+			}
+
+			// We re-call EvaluateSignedData, especially for the size filter, to ensure that the transaction we
+			// just constructed is not too large for our consenter. It additionally reapplies the signature
+			// check, which although not strictly necessary, is a good sanity check, in case the router
+			// has not been configured with the right cert material. The additional overhead of the signature
+			// check is negligible, as this is the re-config path and not the normal path.
+			err = policy.EvaluateSignedData(envelopeAsSignedData)
+			if err != nil {
+				return fmt.Errorf("signature did not satisfy policy %s", policies.ChannelWriters)
+			}
+		}
```

Comment:
This should be implemented above the filters, in the router "process-message" flow